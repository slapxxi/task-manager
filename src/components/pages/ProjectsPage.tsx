import { createProject, getProjectProgress } from '@lib';
import Breadcrumbs from '@local/components/Breadcrumbs/Breadcrumbs';
import Button from '@local/components/Button/Button';
import ProgressCheckbox from '@local/components/ProgressCheckbox/ProgressCheckbox';
import { useStore } from '@local/hooks';
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './styles.css';

function ProjectsPage() {
  const { getProjects, actions } = useStore();
  const projects = getProjects();

  return (
    <div className={styles.projectsPage}>
      <header className={styles.toolbar}>
        <Breadcrumbs />
      </header>
      <h1 className={styles.title}>Projects</h1>
      <>
        <ul className={styles.list}>
          {projects.map((p) => (
            <li key={p.id} className={styles.listItem}>
              <ProgressCheckbox progress={getProjectProgress(p)} />
              <Link to={`/projects/${p.id}`} className={styles.link}>
                {p.name}
              </Link>
            </li>
          ))}
        </ul>
        <Button
          onClick={() => actions.updateProject(createProject({ name: 'Untitled' }))}
        >
          Create Project
        </Button>
      </>
    </div>
  );
}

export default ProjectsPage;
