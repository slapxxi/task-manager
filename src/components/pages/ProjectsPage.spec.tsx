import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render } from 'react-testing-library';
import ProjectsPage from './ProjectsPage';

jest.mock('@local/components/IconSystem/IconSystem');

it('renders', () => {
  const { container } = render(
    <Router>
      <ProjectsPage />
    </Router>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
