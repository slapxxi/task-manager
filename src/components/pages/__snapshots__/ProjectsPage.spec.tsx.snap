// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  class="projectsPage"
>
  <header
    class="toolbar"
  >
    <ul
      class="container"
    >
      <li
        class="item"
      >
        <a
          aria-current="page"
          class="link active"
          href="/"
        >
          <div
            name="inbox"
            size="20"
          >
            icon-
            inbox
          </div>
        </a>
      </li>
    </ul>
  </header>
  <h1
    class="title"
  >
    Projects
  </h1>
  <ul
    class="list"
  />
  <button
    class="sc-bdVaJa fFWIpk"
  >
    Create Project
  </button>
</div>
`;
